awscli/clidocs.py

```diff
@@ -716,3 +730,62 @@ def _line_has_tag(self, line):
 
     def doc_subitems_start(self, help_command, **kwargs):
         pass
+
+
+class GlobalsRenderer:
+    """
+    Renderer used to generate global options for the
+    Synopsis and Global Options sections.
+    """
+
+    def __init__(self, driver):
+        self._driver = driver
+        self._global_options = None
+        self._global_synopsis = None
+
+    @property
+    def global_options(self):
```

Comment:
Then for the public methods, I think we can stick to two public methods and get rid of the `write_global_options` and `write_global_synopsis` methods:
```python
def doc_global_options():
    ...
def doc_global_synopsis():
    ...
```
These methods will just output the documentation string for both the global options and its synopsis. Then the script can take the output and write it to the desired file. The main reason, I'm suggesting it is:
1. It narrows the public interface and makes this abstraction strictly around generating what the content of these sections should be.
2. It makes it more straightforward to test. We can just call these methods and compare it to the content in the `examples` directory instead of having to publicly expose the properties and compare against those.